### ISSUE TITLE
[S10/S10+/S10E] Bluetooh/NFC not working

### ISSUE THREAD

**Issue by uvitor** [phhusson/treble_experimentations#517]
Here is the logcat -b all when i try to open bluetooth/NFC on settings. You can notice from time to time comes an error: "BluetoothAdapter: Bluetooth binder is null"

05-09 22:10:17.414  5545  5646 E LightsService: Light requested not available on this device. 2
05-09 22:10:17.497  8359  8359 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@4d94d2d
05-09 22:10:17.498  5545  5559 I sysui_count: [window_time_0,22]
05-09 22:10:17.498  5545  5559 I sysui_multi_action: [757,803,799,window_time_0,802,22]
05-09 22:10:17.498  5545  5560 I dvm_lock_sample: [system_server,1,android.ui,6,ActivityManagerService.java,4903,void com.android.server.am.ActivityManagerService.dispatchUidsChanged(),-,8165,void com.android.server.am.ActivityManagerService.activityPaused(android.os.IBinder),6]
05-09 22:10:17.513  8359  8359 D SettingsActivity: Starting onCreate
05-09 22:10:17.529  8359  8359 D SettingsActivity: Starting to set activity title
05-09 22:10:17.530  8359  8359 D SettingsActivity: Done setting title
05-09 22:10:17.530  8359  8359 D SettingsActivity: Switching to fragment com.android.settings.connecteddevice.ConnectedDeviceDashboardFragment
05-09 22:10:17.531  8359  8359 E c
05-09 22:10:17.533  8359  8359 E BluetoothAdapter: Bluetooth binder is null
05-09 22:10:17.533  8359  8359 D PrefCtrlListHelper: Could not find Context-only controller for pref: com.android.settings.connecteddevice.AddDevicePreferenceController
05-09 22:10:17.534  8359  8359 E BluetoothAdapter: Bluetooth binder is null
05-09 22:10:17.534  8359  8359 D PrefCtrlListHelper: Could not find Context-only controller for pref: com.android.settings.connecteddevice.PreviouslyConnectedDevicePreferenceController
05-09 22:10:17.534  8359  8359 D PrefCtrlListHelper: Could not find Context-only controller for pref: com.android.settings.connecteddevice.AdvancedConnectedDeviceController
05-09 22:10:17.536  8359  8359 E BluetoothAdapter: Bluetooth binder is null
05-09 22:10:17.537  8359  8359 I chatty  : uid=1000(system) com.android.settings identical 1 line
05-09 22:10:17.539  8359  8359 E BluetoothAdapter: Bluetooth binder is null
05-09 22:10:17.547  8359  8359 D ConnectedDeviceFrag: NO dashboard tiles for ConnectedDeviceFrag
05-09 22:10:17.547  8359  8359 D SettingsActivity: Executed frag manager pendingTransactions
05-09 22:10:17.548  8359  8359 I am_on_create_called: [0,com.android.settings.Settings$ConnectedDeviceDashboardActivity,performCreate]
05-09 22:10:17.555  8359  8359 I am_on_start_called: [0,com.android.settings.Settings$ConnectedDeviceDashboardActivity,handleStartActivity]
05-09 22:10:17.566     0     0 I [2:  kworker/u18:1:14069] [SSP]: MSG From MCU - Sensor_ACC--9,281,4148,60,20 ldo_toggle :0
05-09 22:10:17.556  8359  8359 D AndroidRuntime: Shutting down VM
05-09 22:10:17.558  8359  8359 E AndroidRuntime: FATAL EXCEPTION: main
05-09 22:10:17.558  8359  8359 E AndroidRuntime: Process: com.android.settings, PID: 8359
05-09 22:10:17.558  8359  8359 E AndroidRuntime: java.lang.RuntimeException: Failed to call observer method
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:225)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:193)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:185)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:355)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:293)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:138)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:124)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settingslib.core.lifecycle.ObservablePreferenceFragment.onStart(ObservablePreferenceFragment.java:79)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settings.dashboard.DashboardFragment.onStart(DashboardFragment.java:136)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.Fragment.performStart(Fragment.java:2548)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1334)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1576)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1637)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.FragmentManagerImpl.dispatchMoveToState(FragmentManager.java:3046)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.FragmentManagerImpl.dispatchStart(FragmentManager.java:3003)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.FragmentController.dispatchStart(FragmentController.java:193)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.Activity.performStart(Activity.java:7173)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:193)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6718)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
05-09 22:10:17.558  8359  8359 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.android.settingslib.bluetooth.LocalBluetoothManager.setForegroundActivity(android.content.Context)' on a null object reference
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settings.bluetooth.BluetoothDeviceUpdater.registerCallback(BluetoothDeviceUpdater.java:88)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settings.connecteddevice.AvailableMediaDeviceGroupController.onStart(AvailableMediaDeviceGroupController.java:61)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settingslib.core.lifecycle.Lifecycle.onStart(Lifecycle.java:120)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settingslib.core.lifecycle.Lifecycle.access$100(Lifecycle.java:54)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at com.android.settingslib.core.lifecycle.Lifecycle$LifecycleProxy.onLifecycleEvent(Lifecycle.java:218)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        at android.arch.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:221)
05-09 22:10:17.558  8359  8359 E AndroidRuntime:        ... 30 more
05-09 22:10:17.558  5545 11302 I am_crash: [8359,0,com.android.settings,952647237,java.lang.NullPointerException,Attempt to invoke virtual method 'void com.android.settingslib.bluetooth.LocalBluetoothManager.setForegroundActivity(android.content.Context)' on a null object reference,BluetoothDeviceUpdater.java,88]
05-09 22:10:17.561  5545 11302 W ActivityManager:   Force finishing activity com.android.settings/.Settings$WifiSettingsActivity
05-09 22:10:17.561  5545 11302 I am_finish_activity: [0,168459713,19,com.android.settings/.Settings$WifiSettingsActivity,force-crash]
05-09 22:10:17.562  5545 11302 I am_destroy_activity: [0,168459713,19,com.android.settings/.Settings$WifiSettingsActivity,finish-imm:finishActivityLocked]
05-09 22:10:17.564  5545 11302 W ActivityManager:   Force finishing activity com.android.settings/.Settings$ConnectedDeviceDashboardActivity
05-09 22:10:17.564  5545 11302 I am_finish_activity: [0,174077690,11,com.android.settings/.Settings$ConnectedDeviceDashboardActivity,force-crash]
05-09 22:10:17.565  5545 11302 I am_pause_activity: [0,174077690,com.android.settings/.Settings$ConnectedDeviceDashboardActivity,userLeaving=false]
05-09 22:10:17.566  5545 11302 W ActivityManager:   Force finishing activity com.android.settings/.Settings
05-09 22:10:17.566  5545 11302 I am_finish_activity: [0,33118018,11,com.android.settings/.Settings,force-crash]
05-09 22:10:17.566  5545 11302 I am_focused_stack: [0,0,4,finishActivity adjustFocus]
05-09 22:10:17.566  5545 11302 I wm_task_moved: [7,0,2147483647]
05-09 22:10:17.567  5545  5559 I dvm_lock_sample: [system_server,1,ActivityManager,7,TaskChangeNotificationController.java,152,void com.android.server.am.TaskChangeNotificationController$MainHandler.handleMessage(android.os.Message),AppErrors.java,418,void com.android.server.am.AppErrors.crashApplicationInner(com.android.server.am.ProcessRecord, android.app.ApplicationErrorReport$CrashInfo, int, int),7]
05-09 22:10:17.568  5545  5559 I sysui_count: [window_time_0,0]
05-09 22:10:17.568  5545  5559 I sysui_multi_action: [757,803,799,window_time_0,802,0]
05-09 22:10:17.568  5545  5559 I sysui_count: [window_time_0,0]
05-09 22:10:17.568  5545  5559 I sysui_multi_action: [757,803,799,window_time_0,802,0]
05-09 22:10:17.568  5545  5559 I sysui_count: [window_time_0,0]
05-09 22:10:17.568  5545  5559 I sysui_multi_action: [757,803,799,window_time_0,802,0]
05-09 22:10:17.568  5545  5559 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
05-09 22:10:17.568  5545  5559 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
05-09 22:10:17.572  5545 11302 I sysui_action: [316,-1]
05-09 22:10:17.572  5545 11302 I sysui_multi_action: [757,316,758,4,759,-1]
05-09 22:10:17.577  8359  8359 I Process : Sending signal. PID: 8359 SIG: 9
05-09 22:10:17.652  5545  5646 W InputDispatcher: channel 'd1d0368 com.android.settings/com.android.settings.Settings$WifiSettingsActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
05-09 22:10:17.652  5545  5646 E InputDispatcher: channel 'd1d0368 com.android.settings/com.android.settings.Settings$WifiSettingsActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
05-09 22:10:17.652  5545  5646 W InputDispatcher: channel '88c78c8 com.android.settings/com.android.settings.Settings (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
05-09 22:10:17.652  5545  5646 E InputDispatcher: channel '88c78c8 com.android.settings/com.android.settings.Settings (server)' ~ Channel is unrecoverably broken and will be disposed!
05-09 22:10:17.660  6162  6164 I adbd    : initializing functionfs
05-09 22:10:17.660  6162  6164 I adbd    : opening control endpoint /dev/usb-ffs/adb/ep0
05-09 22:10:17.669  5545  5657 W WifiService: Client connection lost with reason: 4
05-09 22:10:17.669  5545  5657 W WifiService: Client connection lost with reason: 4
05-09 22:10:17.669  5545  7281 I WindowManager: WIN DEATH: Window{88c78c8 u0 com.android.settings/com.android.settings.Settings}
05-09 22:10:17.669  5545  7281 W InputDispatcher: Attempted to unregister already unregistered input channel '88c78c8 com.android.settings/com.android.settings.Settings (server)'
05-09 22:10:17.671  5231  5231 I Zygote  : Process 8359 exited due to signal (9)
05-09 22:10:17.672  5545  7696 I ActivityManager: Process com.android.settings (pid 8359) has died: vis  +99TOP
05-09 22:10:17.672  5545  7696 I am_proc_died: [0,8359,com.android.settings,199,2]
05-09 22:10:17.672  5545  9287 I WindowManager: WIN DEATH: Window{d1d0368 u0 com.android.settings/com.android.settings.Settings$WifiSettingsActivity}
05-09 22:10:17.672  5545  9287 W InputDispatcher: Attempted to unregister already unregistered input channel 'd1d0368 com.android.settings/com.android.settings.Settings$WifiSettingsActivity (server)'
05-09 22:10:17.675  5545  5562 W libprocessgroup: kill(-8359, 9) failed: No such process
05-09 22:10:17.675  5545  5562 I libprocessgroup: Successfully killed process cgroup uid 1000 pid 8359 in 0ms
05-09 22:10:17.678  5292  5335 W SurfaceFlinger: Attempting to set client state on removed layer: com.android.settings/com.android.settings.Settings#0
05-09 22:10:17.678  5292  5335 W SurfaceFlinger: Attempting to destroy on removed layer: com.android.settings/com.android.settings.Settings#0
05-09 22:10:17.679  5545  7696 I wm_task_removed: [11,removeAppToken: last token]
05-09 22:10:17.679  5545  7696 I am_remove_task: [11,4]
05-09 22:10:17.679  5545  7696 I wm_task_removed: [11,removeTask]
05-09 22:10:17.681  5292  5334 W SurfaceFlinger: Attempting to destroy on removed layer: AppWindowToken{d629490 token=Token{ecc7353 ActivityRecord{1f95742 u0 com.android.settings/.Settings t11}}}#0
05-09 22:10:17.682  5545  7696 I wm_stack_removed: 4
05-09 22:10:17.684  5545  7696 I wm_task_removed: [19,removeAppToken: last token]
05-09 22:10:17.684  5545  7696 I am_remove_task: [19,12]
05-09 22:10:17.684  5545  7696 I wm_task_removed: [19,removeTask]
05-09 22:10:17.686  5292  5334 W SurfaceFlinger: Attempting to destroy on removed layer: AppWindowToken{7f1c4a7 token=Token{d0e8966 ActivityRecord{a0a7dc1 u0 com.android.settings/.Settings$WifiSettingsActivity t19}}}#0
05-09 22:10:17.687  5545  7696 I wm_stack_removed: 12
05-09 22:10:17.694  5545  7696 I am_set_resumed_activity: [0,com.android.launcher3/.Launcher,resumeTopActivityInnerLocked]
05-09 22:10:17.699  5545  7696 I am_resume_activity: [0,31636405,7,com.android.launcher3/.Launcher]
05-09 22:10:17.705  5545  6207 I dvm_lock_sample: [system_server,1,Binder:5545_8,33,ActivityManagerService.java,22373,android.app.ActivityManager$StackInfo com.android.server.am.ActivityManagerService.getFocusedStackInfo(),-,1890,void com.android.server.am.ActivityManagerService$AppDeathRecipient.binderDied(),33]
05-09 22:10:17.706  5545  5567 I dvm_lock_sample: [system_server,1,android.display,29,ActivityManagerService.java,26479,void com.android.server.am.ActivityManagerService$LocalService.notifyAppTransitionFinished(),-,1890,void com.android.server.am.ActivityManagerService$AppDeathRecipient.binderDied(),29]
05-09 22:10:17.707  5545  5947 I dvm_lock_sample: [system_server,1,Binder:5545_6,29,ActivityManagerService.java,10979,android.app.ActivityManager$TaskSnapshot com.android.server.am.ActivityManagerService.getTaskSnapshot(int, boolean),-,1890,void com.android.server.am.ActivityManagerService$AppDeathRecipient.binderDied(),29]
05-09 22:10:17.708  5545  5567 W ActivityManager: setHasOverlayUi called on unknown pid: 8359

**Comment by phhusson** [comment]
Hum I'm a bit suprised, people have been using GSI on S10 for 2 weeks now, and you're the first report of broken bluetooth?
(NFC I can understand, noone uses NFC)

**Comment by penn5** [comment]
Bluetooth : binder is null
means that BluetoothApp is not bound to framework

**Comment by uvitor** [comment]
> Hum I'm a bit suprised, people have been using GSI on S10 for 2 weeks now, and you're the first report of broken bluetooth?
> (NFC I can understand, noone uses NFC)

Actually a few of us S10 users who tests GSIs reported on the telegram group (me included couple of weeks ago), but i know thats hard to follow all that comunity reporting all sorts of issues...
So, its not just me. It doesnt work on yours and all other GSIs tested by me. If you need me another kind of log or testing let me know, im here to help. Thank you very much.

**Comment by phhusson** [comment]
I'm guessing your issue is https://github.com/phhusson/treble_experimentations/issues/522, but your logs doesn't show.
Could you give a full logcat, not just an extract, and as an attached file.
Also, could you attach your stock ROM's /system/etc/selinux and /vendor/etc/selinux?

**Comment by uvitor** [comment]
Is there some action you want me to do to show up on logcat? Because that previous logcat was when i tried to open bluetooth in settinngs. Here are the selinux folders 

[selinux(system).zip](https://github.com/phhusson/treble_experimentations/files/3200331/selinux.system.zip)
[selinux(vendor).zip](https://github.com/phhusson/treble_experimentations/files/3200332/selinux.vendor.zip)

**Comment by phhusson** [comment]
Ok issue doesn't seem to be same as #522 grmbl.
For logcats, do the following:
- reboot your phone
- directly go to bluetooth settings and try to enable bluetooth
- from a computer do `adb logcat -b all -d > logs.txt`
- attach the generated logs.txt to this issue

**Comment by elmkzgirxp** [comment]
Did all the steps. Here is the log for v114.

[logs.txt](https://github.com/phhusson/treble_experimentations/files/3268922/logs.txt)

**Comment by penn5** [comment]
auditd  : type=1400 audit(0.0:20): avc: denied { mounton } for comm="mount" path="/mnt/phh" dev="tmpfs" ino=26720 scontext=u:r:phhsu_daemon:s0 tcontext=u:object_r:tmpfs:s0 tclass=dir permissive=0
Excuse me what the fuck

**Comment by phhusson** [comment]
@penn5 it's likely Samsung kernels ignore permissive contexts

**Comment by phhusson** [comment]
```
06-09 04:50:40.970  5614  5614 I SystemServer: No Bluetooth Service (Bluetooth Hardware Not Present)
```

Mmmmkay...

Please test with Google's GSI:

https://ci.android.com/builds/submitted/5658405/aosp_arm64-eng/latest/aosp_arm64-img-5658405.zip

**Comment by elmkzgirxp** [comment]
I couldn't get any of the eng builds to flash properly (can't mount /system) so I just used a userdebug build on aosp-pie-gsi branch.

Build 5657968
[log1.txt](https://github.com/phhusson/treble_experimentations/files/3291056/log1.txt)

**Comment by phhusson** [comment]
You mean it doesn't work on Google's gsi either?

**Comment by elmkzgirxp** [comment]
Yep. Same crashes.

**Comment by ktshikotshi** [comment]
- Bluetooth` works after installing the Pix3lify magisk module, I don't know why but it works
- It seems NFC was never broken, It is switch on by default and can be switched off using the quick settings

*Tested on `Phh v114` and `Pixel Experiance(PE-ARM-A-2019-03-24)`

**Comment by phhusson** [comment]
Link?

**Comment by ktshikotshi** [comment]
[pix3lify-unity-magisk-module](https://magiskroot.net/pix3lify-unity-magisk-module/)

Do you want me to do a logcat as well?

**Comment by elmkzgirxp** [comment]
@ktshikotshi is right.

The `handheld_core_hardware` vendor permissions file has `<feature name="android.hardware.bluetooth" />` commented out along with other sensors such as proximity and light. Uncommenting this line should fix it.

**Comment by ktshikotshi** [comment]
@elmkzgirxp @phhusson I can confirm that removing the comment on `handheld_core_hardware` fixes the bluetooth, i added `android.hardware.sensor.proximity.xml` to try fixing the proximity sensor, but i get the following on my log
`07-02 18:03:27.880 12205 12205 D PageDetailActivity.registerSensorListeners: type = 65596 / name = Touch Proximity Sensor
07-02 18:03:27.881  5151  5151 D Sensors : M requested delay = 200000000, modified delay = 0
07-02 18:03:27.881  5151  5151 I nanohub : mergeDirectReportRequest - activate = 0 normal.rate = 0
07-02 18:03:27.881  5151  5151 I Sensors : batch(27) - sec_touchproximity try to batching with 0
07-02 18:03:27.881  5151  5151 E Sensors : TouchProximitySensor::write_sysfs_int: open failed: -13(Permission denied)
07-02 18:03:27.881  5151  5151 I Sensors : TouchProximitySensor::enable: 1 0` 
/----------
for Touch proximity sensor and 
`07-02 18:03:27.872 12205 12205 D PageDetailActivity.registerSensorListeners: type = 65589 / name = Proximity Pocket
07-02 18:03:27.873  5151  5151 D Sensors : M requested delay = 200000000, modified delay = 0
07-02 18:03:27.873  5151  5151 I nanohub : mergeDirectReportRequest - activate = 1 normal.rate = 0
07-02 18:03:27.873  5151  5151 I Sensors : batch(22) - proximity_pocket try to batching with 0
07-02 18:03:27.873  5151  5151 I Sensors : ProximityPocket old sensor_state 33604151, new sensor_state : 1142494904887 en : 1
07-02 18:03:27.873  5151  5151 E Sensors : ProximityPocket::enable: write failed: -22(Invalid argument)
07-02 18:03:27.874  5497  9938 E SensorService: Error activating sensor 11 (Invalid argument)`
/----------
 for proximity pocket sensor
